src/sources/http.rs

```diff
@@ -20,41 +17,47 @@ use tokio_util::codec::Decoder;
 use warp::http::{HeaderMap, HeaderValue, StatusCode};
 
 #[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(default)]
```

Comment:
I think this (plus the `Default` trait implementation below) will have the undesired effect of not requiring the user to specify the `address` field in their config. We currently require this field.

On the other hand, I could see trying to choose a sensible default for it like you did here. I might suggest something other than `:80` as that is likely to be blocked.

For now, I'd recommend just replacing the `#[serde(default)]`s below and removing it from the struct. You can provide a default value for the field with something like:

https://github.com/timberio/vector/blob/c79545c74bb981c66e1082c12d635240c777aefd/src/sources/syslog.rs#L40-L41

https://github.com/timberio/vector/blob/c79545c74bb981c66e1082c12d635240c777aefd/src/sources/syslog.rs#L64-L66

Reply:
I think not changing the default behaviour should be favoured when possible, I followed your suggestion to keep the current mandatory behaviour for the `address` setting and rely on function to provided other default values.